sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/environment.py

```diff
@@ -317,13 +321,14 @@ def __eq__(self, other) -> bool:
     def __ne__(self, other) -> bool:
         return not self.__eq__(other)
 
-    def _generate_anonymous_name_version(self, source: str, conda_file: str = None):
+    def _generate_anonymous_name_version(self, source: str, conda_file: str = None, inference_config: Dict = None):
         hash_str = ""
         if source == "image":
-            if not conda_file:
-                hash_str = hash_str.join(get_md5_string(self.image))
-            else:
-                hash_str = hash_str.join(get_md5_string(self.image)).join(get_md5_string(conda_file))
+            hash_str = hash_str.join(get_md5_string(self.image))
+            if inference_config:
+                hash_str = hash_str.join(get_md5_string(yaml.dump(inference_config, sort_keys=True)))
```

Comment:
(not blocking) - I am wondering if we should create a helper function to do safe md5 hash for dictionaries. I'm just worried someone forgets to do this then we create inconsistent hashes

Reply:
This is a good thought, curious when you say someone forgets this, do you mean end users trying to do it by themselves, instead of using this part of SDK?

Reply:
Another Azure ML dev, not an end user.

Reply:
Ok I see, I am thinking it would be useful if we allow dev's to use the same logic. I will open an item to see if this would be a requirement and we can take it from there.